grug/types/src/coin_pair.rs

```diff
@@ -160,6 +160,58 @@ impl CoinPair {
             },
         ]))
     }
+
+    /// Return true if the coin pair has the given denom.
+    pub fn has(&self, denom: &Denom) -> bool {
+        self.first().denom == denom || self.second().denom == denom
+    }
+
+    /// Return the amount of the given denom in the coin pair.
+    pub fn amount_of(&self, denom: &Denom) -> Uint128 {
+        if self.first().denom == denom {
+            *self.first().amount
+        } else {
+            *self.second().amount
+        }
+    }
```

Comment:
i think you need to account for the case where `denom` equals neither the two denoms in the `CoinPair`

Reply:
good catch. will return a result instead

Reply:
fixed